webapp/channels/src/components/suggestion/suggestion_box/suggestion_box.jsx

```diff
@@ -564,7 +564,38 @@ export default class SuggestionBox extends React.PureComponent {
         return this.state.items.some((item) => !item.loading);
     };
 
+    confirmPretext = () => {
+        const textbox = this.getTextbox();
+        const pretext = textbox.value.substring(0, textbox.selectionEnd).toLowerCase();
+
+        if (this.pretext !== pretext) {
+            this.handlePretextChanged(pretext);
+        }
+    };
+
+    handleKeyPress = (e) => {
+        this.confirmPretext();
+        if (this.props.onKeyPress) {
+            this.props.onKeyPress(e);
+        }
+    };
+
+    handleKeyUp = (e) => {
+        this.confirmPretext();
+        if (this.props.onKeyUp) {
+            this.props.onKeyUp(e);
+        }
+    };
+
+    handleMouseUp = (e) => {
+        this.confirmPretext();
+        if (this.props.onMouseUp) {
+            this.props.onMouseUp(e);
+        }
+    };
+
     handleKeyDown = (e) => {
+        this.confirmPretext();
```

Comment:
Do we need to do this on key press, up, and down? I would've expected just one of those to be sufficient since we're now checking if the pretext changed potentially 4 times per key press while the user is typing

Reply:
Hopefully, on up is enough. The main problem here is when the caret position is updated in the textbox. From my tests, if I remember correctly, it is not on key down. And may depend on the browser. And I haven't test what happens / what is expected if you just keep the key down without releasing it.

I will do a few more tests around that.

Reply:
Yeah, that sounds good to me. I'd be good just waiting until `keyup` since I don't think a regular user would actually want to see results for every character as they hold down an arrow key

Reply:
The problem with that is that you can "force the bug" (but you still can do it in many other ways) by holding the back button (the suggestions won't disappear because you have not changed the pretext) and hit enter while still have the back button clicked.

But that seems a corner case we don't need to worry about.

Reply:
Very good point. Maybe we could split the show/hide logic for the autocomplete so that we can hide it on keydown and then show new results on keyup, but that's likely complicating things. As you said, it's a corner case, so I'm good with leaving that out for now